src/cli/cmd/init/mod.rs

```diff
@@ -0,0 +1,311 @@
+};
+use inquire::{Confirm, MultiSelect, Select, Text};
+use serde::Serialize;
+use serde_json::Value;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use std::process::ExitCode;
+
+use super::{CommandExecute, FhError};
+
+const CARGO_TOOLS: &[&str] = &[
+    "audit", "bloat", "cross", "edit", "outdated", "udeps", "watch",
+];
+
+const NODE_VERSIONS: &[&str] = &["18", "16", "14"];
+const JS_PACKAGING_TOOLS: &[&str] = &["pnpm", "yarn"];
+
+const GO_VERSIONS: &[&str] = &["20", "19", "18", "17"];
+
+const SYSTEMS: &[&str] = &[
+    "x86_64-linux",
+    "aarch64-linux",
+    "x86_64-darwin",
+    "aarch64-darwin",
+];
+
+/// Create a new flake.nix using an interactive initializer.
+#[derive(Parser)]
+pub(crate) struct InitSubcommand {
+    #[clap(long, short, default_value = ".")]
+    root: PathBuf,
+
+    #[clap(long, short, default_value = "./flake.nix")]
+    output: PathBuf,
+}
+
+#[async_trait::async_trait]
+impl CommandExecute for InitSubcommand {
+    async fn execute(self) -> Result<ExitCode> {
+        let mut inputs: HashMap<String, String> = HashMap::new();
+        let mut dev_shell_packages: Vec<String> = Vec::new();
+        let mut dev_shells: HashMap<String, DevShell> = HashMap::new();
+        let mut overlay_refs: Vec<String> = Vec::new();
+        let mut overlay_attrs: HashMap<String, String> = HashMap::new();
+        let mut packages: HashMap<String, String> = HashMap::new();
+
+        if self.output.exists() && !Prompt::bool("A flake.nix already exists in the current directory. Would you like to overwrite it?")? {
+            println!("Exiting. Let's a build a new flake soon, though :)");
+            return Ok(ExitCode::SUCCESS);
+        }
+
+        println!("Let's build a Nix flake!");
+
+        let project = Project::new(self.root);
+        let description = Prompt::maybe_string("A description for your flake")?;
+
+        let systems =
+            MultiSelect::new("Which systems would you like to support?", SYSTEMS.to_vec())
+                .prompt()?
+                .iter()
+                .map(|s| String::from(*s))
+                .collect();
+
+        let nixpkgs = Confirm::new("Do you want to include a Nixpkgs input?")
+            .with_default(false)
+            .prompt()?;
+
+        if nixpkgs {
+            inputs.insert(
+                String::from("nixpkgs"),
+                String::from("https://flakehub.com/f/NixOS/nixpkgs/*.tar.gz"), // TODO: make this more granular
+            );
+        }
+
+        if Prompt::bool(
+            "Do you want to add the commonly used Nix formatter (nixpkgs-fmt) to your environment?",
+        )? {
+            dev_shell_packages.push(String::from("nixpkgs-fmt"));
+        }
+
+        // Rust projects
+        if project.maybe_rust() && Prompt::bool("This seems to be a Rust project. Would you like to initialize your flake with built-in Rust dependencies?")? {
+            // Add Rust overlay
```

Comment:
Would you consider [fenix](https://github.com/nix-community/fenix) here? :)

The main advantage over rust-overlay is that it is a lot smaller, and the packages are cached in nix-community cachix. The drawback is that using arbitrary versions (instead of the latest stable/beta/nightly) would require a hash or a flake input.

Another option is to have the choice between these 2 and just regular rust from nixpkgs

Disclaimer: I wrote fenix

Reply:
I'm fine with switching, but I can't quite seem to get `fromToolchainFile` working correctly. Could you provide an example? In general, I find the usage section of the docs pretty confusing, as I'm not sure if I need to use `prev.fenix.fromToolchainFile` or `inputs.fenix.fromToolchainFile` or `fenix.packages.fromToolchainFile` or etc.

Reply:
Never mind, I figured it out 😄 But on second though, I think a (fake hash -> provide correct hash) workflow is more hands-on than I want to do for this first iteration of the tool, so I'm going to stick with `rust-overlay` as the most auto-magical option available. But in the future, I'd definitely be open to more nuanced approaches.